packages/js/email-editor/src/components/personalization-tags/test/category-menu.spec.tsx

```diff
@@ -0,0 +1,108 @@
+import '../../__mocks__/setup-shared-mocks';
+/* eslint-disable @woocommerce/dependency-group -- because we import mocks first, we deactivate this rule to avoid es lint errors */
+
+/**
+ * External dependencies
+ */
+import { render, screen, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
+/**
+ * Internal dependencies
+ */
+import { CategoryMenu } from '../category-menu';
+
+jest.mock( '@wordpress/components', () => ( {
+	MenuGroup: ( props: React.HTMLAttributes< HTMLDivElement > ) => (
+		<div data-testid="menu-group" { ...props } />
+	),
+	MenuItem: ( props: React.ButtonHTMLAttributes< HTMLButtonElement > ) => (
+		<button role="menuitem" { ...props } />
+	),
+} ) );
+
+describe( 'CategoryMenu', () => {
+	const groupedTags = {
+		Marketing: [],
+		Promotions: [],
+		'Customer Data': [],
+	};
+	const onCategorySelect = jest.fn();
+
+	beforeEach( () => {
+		onCategorySelect.mockClear();
+	} );
+
+	it( 'should render the "All" menu item and all categories', () => {
+		render(
+			<CategoryMenu
+				groupedTags={ groupedTags }
+				activeCategory={ null }
+				onCategorySelect={ onCategorySelect }
+			/>
+		);
+
+		expect( screen.getByText( 'All' ) ).toBeInTheDocument();
+		expect( screen.getByText( 'Marketing' ) ).toBeInTheDocument();
+		expect( screen.getByText( 'Promotions' ) ).toBeInTheDocument();
+		expect( screen.getByText( 'Customer Data' ) ).toBeInTheDocument();
+	} );
+
+	it( 'should call onCategorySelect when "All" is clicked', () => {
+		render(
+			<CategoryMenu
+				groupedTags={ groupedTags }
+				activeCategory={ 'Marketing' }
+				onCategorySelect={ onCategorySelect }
+			/>
+		);
+
+		fireEvent.click( screen.getByText( 'All' ) );
+		expect( onCategorySelect ).toHaveBeenCalledWith( null );
+	} );
+
+	it( 'should call onCategorySelect with correct category', () => {
+		render(
+			<CategoryMenu
+				groupedTags={ groupedTags }
+				activeCategory={ null }
+				onCategorySelect={ onCategorySelect }
+			/>
+		);
+
+		fireEvent.click( screen.getByText( 'Promotions' ) );
+		expect( onCategorySelect ).toHaveBeenCalledWith( 'Promotions' );
+	} );
+
+	it( 'should apply active class to active category', () => {
+		render(
+			<CategoryMenu
+				groupedTags={ groupedTags }
+				activeCategory={ 'Customer Data' }
+				onCategorySelect={ onCategorySelect }
+			/>
+		);
+
+		const activeItem = screen.getByText( 'Customer Data' );
+		expect( activeItem ).toHaveClass(
+			'woocommerce-personalization-tags-modal-menu-item-active'
+		);
+	} );
+
+	it( 'should render separators between categories', () => {
+		render(
+			<CategoryMenu
+				groupedTags={ groupedTags }
+				activeCategory={ null }
+				onCategorySelect={ onCategorySelect }
+			/>
+		);
+
+		const separators = screen.getAllByTestId(
+			'woocommerce-personalization-tags-modal-menu-separator'
+		);
+		// 1 after "All" + 2 between 3 categories
+		expect( separators ).toHaveLength( 3 );
+	} );
```

Comment:
In this case it feels like we are testing something visual rather than logic. Should we add this type of checks? I don't have a strong opinion. I is just a thought. I got to it because I saw we needed to add data-testid for this one.
Do you think we need to test this?